spring-kafka/src/main/java/org/springframework/kafka/config/RetryTopicConfigurationSupport.java

```diff
@@ -0,0 +1,339 @@
+	 * by non-blocking retries' configuration.
+	 * @param customizersConfigurer a {@link CustomizersConfigurer}.
+	 */
+	protected void configureCustomizers(CustomizersConfigurer customizersConfigurer) {
+	}
+
+	/**
+	 * Return a global {@link DestinationTopicResolver} for resolving
+	 * the {@link org.springframework.kafka.retrytopic.DestinationTopic}
+	 * to which a given {@link org.apache.kafka.clients.consumer.ConsumerRecord}
+	 * should be sent for retry.
+	 *
+	 * To configure it, consider overriding one of these other more
+	 * fine-grained methods:
+	 * <ul>
+	 * <li>{@link #configureNonBlockingRetries} for configuring non-blocking retries.
+	 * <li>{@link #customizeDestinationTopicResolver} for further customizing the component.
+	 * <li>{@link #createComponentFactory} for providing a subclass instance.
+	 * </ul>
+	 *
+	 * @param context the {@link ApplicationContext}.
+	 * @return the instance.
+	 */
+	@Bean(name = RetryTopicBeanNames.DESTINATION_TOPIC_RESOLVER_BEAN_NAME)
+	public DestinationTopicResolver destinationTopicResolver(ApplicationContext context) {
+		DestinationTopicResolver destinationTopicResolver = this.componentFactory.destinationTopicResolver(context);
+		if (destinationTopicResolver instanceof DefaultDestinationTopicResolver) {
+			DefaultDestinationTopicResolver ddtr = (DefaultDestinationTopicResolver) destinationTopicResolver;
+			NonBlockingRetriesConfigurer configurer = new NonBlockingRetriesConfigurer();
+			configureNonBlockingRetries(configurer);
+			if (configurer.clearDefaultFatalExceptions) {
+				ddtr.setClassifications(new HashMap<>(), true);
+			}
+			if (configurer.addToFatalExceptions != null) {
+				Stream.of(configurer.addToFatalExceptions).forEach(ddtr::addNotRetryableExceptions);
+			}
+			if (configurer.removeFromFatalExceptions != null) {
+				Stream.of(configurer.removeFromFatalExceptions).forEach(ddtr::removeClassification);
+			}
+		}
+		Consumer<DestinationTopicResolver> resolverConsumer = customizeDestinationTopicResolver();
+		Assert.notNull(resolverConsumer, "customizeDestinationTopicResolver must not return null");
+		return destinationTopicResolver;
+	}
+
+	/**
+	 * Override this method to configure the {@link DestinationTopicResolver}.
+	 * @return a {@link DestinationTopicResolver} consumer.
+	 */
+	protected Consumer<DestinationTopicResolver> customizeDestinationTopicResolver() {
+		return dtr -> {
+		};
+	}
+
+	/**
+	 * Override this method if you want to provide a subclass
+	 * of {@link RetryTopicComponentFactory} with different
+	 * component implementations or subclasses.
+	 * @return the instance.
+	 */
+	protected RetryTopicComponentFactory createComponentFactory() {
+		return new RetryTopicComponentFactory();
+	}
+
+	@Deprecated
+	@Bean(name = RetryTopicInternalBeanNames.RETRY_TOPIC_BOOTSTRAPPER)
+	RetryTopicBootstrapper retryTopicBootstrapper(ApplicationContext context) {
+		return new RetryTopicBootstrapper(context, context.getAutowireCapableBeanFactory());
```

Comment:
This is an overhead. 
Consider to use `ApplictionContextAware` and `BeanFactoryAware`.
The framework will inject them respectively when this bean is created.

Perhaps many other features you create via specific factory could be handled also as just beans and application context hooks...

Reply:
Yes, I think I mentioned this somewhere. This specific bean is deprecated and I'm not sure it's worth change anything on it, but there are a couple other beans that should benefit from it.

My initial concern was about changing this for a patch version, but since we're releasing this in a milestone I think we can just go ahead and do it.